midend/removeMiss.cpp

```diff
@@ -26,4 +26,19 @@ const IR::Node* DoRemoveMiss::preorder(IR::Member* expression) {
     return new IR::LNot(hit);
 }
 
+const IR::Node* DoRemoveMiss::preorder(IR::IfStatement* statement) {
+    if (!TableApplySolver::isMiss(statement->condition, refMap, typeMap))
+        return statement;
+    auto mem = statement->condition->to<IR::Member>();
+    CHECK_NULL(mem);
```

Comment:
I have been fond of [checkedTo](https://github.com/p4lang/p4c/blob/08ef61a007b47b48945f86d60fc47b81b4d0fef1/ir/node.h#L64) for these.

Reply:
Forgot about that one